### PR TITLE
Django 4 upgrade

### PR DESCRIPTION
**Pull request by ukanga**
### Changes / Features implemented

Updating dependences and code to work with Django 4+

### Side effects of implementing this change

**Before submitting this PR for review, please make sure you have:**

  - [ ] Included tests
  - [ ] Updated documentation

Closes #

### CODE REVIEW

File: requirements/azure.pip [hunk 1]
```diff
@@ -22,20 +22,18 @@ cryptography==42.0.5
     # via
     #   -r requirements/azure.in
     #   azure-storage-blob
-django==3.2.25
+django==4.0
```
Comment (KipSigei) [inline requirements/azure.pip:25]: we should update to 4.2 which is currently in LTS, 4.0 ended security support [11 months ago](https://endoflife.date/django)

Comment (ukanga) [reply requirements/azure.pip:25]: Oh, yes. I just remembered that my strategy was to go through all the versions but start with 4.0. I will take this back to the draft.

File: setup.cfg [hunk 2]
```diff
@@ -102,6 +101,7 @@ install_requires =
     # Deprecation tagging
     deprecated
     # Redis cache
+    redis[hiredis]
     django-redis
```
Comment (KipSigei) [inline setup.cfg:105]: Note that in django 4 we dont require the `django-redis` package as its now implemented via `django.core.cache.backends.redis.RedisCache`

Comment (ukanga) [reply setup.cfg:105]: We may need a bit more extensive testing for the switch. I recommend we do that separately and not as part of this upgrade. For more context on `django-redis` [see](https://github.com/jazzband/django-redis/issues/585#issuecomment-1002051473).

**Review by KipSigei — DISMISSED** [review]

**Review by KipSigei — APPROVED** [review]

**Review by KipSigei — APPROVED** [review]

**Pull request merged**